# Electric Motor Temperature Data Summary

This data comes from... 

## Part 1: Data Formatting

In [1]:
# Libraries:
import pandas as pd
import numpy as np

In [2]:
# Import the data:
path = "../data/Electric_Motor_Temp_Data.csv"
electric_data = pd.read_csv(path, header = 0)

First, we check the data types and check for missing values.

In [4]:
# Data types:
electric_data.dtypes

u_q               float64
coolant           float64
stator_winding    float64
u_d               float64
stator_tooth      float64
motor_speed       float64
i_d               float64
i_q               float64
pm                float64
stator_yoke       float64
ambient           float64
torque            float64
profile_id          int64
dtype: object

In [5]:
# Missing values:
print(all(electric_data.isna() == False))
print(all(electric_data.isnull() == False))

True
True


All columns have a numeric data type and there does not appear to be any missing values.

In [6]:
# Check the data set shape
electric_data.shape

(1330816, 13)

The data set contains 1,330,816 observations with 13 variables. The predictor variables include voltage q-component measurement(u_q), coolant temperature(degree celsius), stator winding temperature(degree celsius), voltage d-component measurement(u_d), strator tooth temperature(degrees Celsius), motor speed(rpm), current d-component measurement, current q-component measurement, stator yoke(degree celsius). The response variable is the permanent magnet temperature(pm - degree celsius).